src/hal/src/image.rs

```diff
@@ -56,15 +56,20 @@ impl Extent {
     }
 }
 
-///
+/// An offset into an `Image` used for image-to-image
+/// copy operations.  All offsets are in texels, and
+/// specifying offsets other than 0 for dimensions
+/// that do not exist is undefined behavior -- for
+/// example, specifying a `z` offset of `1` in a
+/// two-dimensional image.
 #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct Offset {
-    ///
+    /// X offset, in texels
```

Comment:
I think we can have a `type TexelCoordinate = i32` to reflect this

Reply:
:+1: